packages/calcite-components/src/utils/number.spec.ts

```diff
@@ -169,3 +173,34 @@ describe("expandExponentialNumberString", () => {
     expect(expandExponentialNumberString("")).toBe("");
   });
 });
+
+describe("addLocalizedTrailingDecimalZeros", () => {
+  locales.forEach((locale) => {
+    it(`add back sanitized trailing decimal zero values - ${locale}`, () => {
+      numberStringFormatter.numberFormatOptions = {
+        locale,
+        // the group separator is different in arabic depending on the numberingSystem
+        numberingSystem: locale === "ar" ? "arab" : "latn",
+        useGrouping: true
+      };
+      const localizedValue = numberStringFormatter.localize("123456.000");
+      const localizedZeroValue = numberStringFormatter.localize("0");
+      const result = `${localizedValue}${numberStringFormatter.decimal}`.padEnd(
+        localizedValue.length + 4,
+        localizedZeroValue
```

Comment:
can you add a few more spec tests for the following numbers:

- `1230000000000000000000000000000.00000000000000000000045000000000000000000000000`
- `-102.02100e1000000`

The regex may need to be adjusted to not consider trailing zeros after an `e`.

Reply:
It would probably be easier to make sure `indexOf("e") === -1` rather than adding a negative lookaheads or something to the regex

Reply:
Good catch, the 0 after the `e` should be ignored.